packages/react-dom-bindings/src/server/ReactDOMServerFormatConfig.js

```diff
@@ -1660,33 +1667,100 @@ function pushStartHead(
   target: Array<Chunk | PrecomputedChunk>,
   preamble: Array<Chunk | PrecomputedChunk>,
   props: Object,
-  tag: string,
   responseState: ResponseState,
 ): ReactNodeList {
-  return pushStartGenericElement(
-    enableFloat ? preamble : target,
-    props,
-    tag,
-    responseState,
-  );
+  if (enableFloat) {
+    let children = null;
+    let innerHTML = null;
+    let includedAttributeProps = false;
+
+    if (!responseState.hasHead) {
+      responseState.hasHead = true;
```

Comment:
`<head>` should not allow any attributes in this mode. It's too flaky in case we flush the preamble before it's discovered so it would have different semantics in different cases depending on data. So it should never have attributes to avoid you accidentally relying on it.

In other words, this can all be much more simplified because in this mode, you never write head when pushing.

You can always write head when you're writing the HTML tag. Whether it's implicit or not.

That means that the whole thing gets much more simplified because the whole preamble is just a boolean. Whether you have html+head or not.